src/gui/transferlistsortmodel.cpp

```diff
@@ -42,6 +42,7 @@ TransferListSortModel::TransferListSortModel(QObject *parent)
     : QSortFilterProxyModel {parent}
 {
     QMetaType::registerComparators<BitTorrent::TorrentState>();
+    setSortRole(TransferListModel::UnderlyingDataRole);
```

Comment:
Did I forget to do this? I was going to...

Reply:
No, it's here:
https://github.com/qbittorrent/qBittorrent/blob/1917064bdeb1bc42509478c0049abbf288b2b1ee/src/gui/transferlistwidget.cpp#L144

Reply:
oh well... then I've no idea why recently there are issues complaining about sorting not working.

Reply:
Maybe they were issues that affected outdated qBittorrent version?

Reply:
>Maybe they were issues that affected outdated qBittorrent version?

It could be. Let's see how it trend for later releases.

Reply:
@Chocobo1 @glassez 
IIRC the recent complaints about sorting relate to subsorting. Normal sorting is working fine. Apparently, previously qBittorrent was subsorting by name, whereas now it is either subsorting by added #, or not at all.

Here, "subsorting by name" means the following: If you sort all torrents by % complete, for example, all torrents with the same % complete will be additionally sorted by name.

If subsorting is done, I think it makes sense to always be done by name.